lib/components/ConfirmDialog/ConfirmDialog.tsx

```diff
@@ -0,0 +1,114 @@
+import { useState } from 'react';
+import { Description, Dialog, DialogPanel, DialogTitle } from '@headlessui/react';
+import { cx } from '../../cva';
+
+type Variant = 'normal' | 'destructive';
+
+type FooterProps =
+  | {
+      footer?: never;
+
+      onConfirm: () => void;
+      confirmText: string;
+      variant?: Variant;
+      cancelText: string;
+    }
+  | {
+      footer: (closeDialog: () => void) => React.ReactNode;
+
+      onConfirm?: never;
+      confirmText?: never;
+      variant?: never;
+      cancelText?: never;
+    };
```

Comment:
Use of "never" should be avoided. Is there also a need for two different types?

Reply:
These types are made so that if `footer` prop is given, then there is no sense to give the other props listed as they are not used and vice versa.
Do not know if there is some other than `never` type to use in these conditional types.

Reply:
How about something like this?

```suggestion
type FooterProps =
  | {
      onConfirm: () => void;
      confirmText: string;
      variant?: Variant;
      cancelText: string;
    }
  | {
      footer: (closeDialog: () => void) => React.ReactNode;
    };
```